compiler/qsc_frontend/src/resolve.rs

```diff
@@ -569,6 +582,68 @@ impl Resolver {
                 scope.tys.insert(Rc::clone(&name.name), id);
                 scope.terms.insert(Rc::clone(&name.name), id);
             }
+            ast::ItemKind::Export(export) => {
+                // resolve the exported item and insert the vec ident into the names table, so we can access it in
+                // lowering
+                for item in export.items() {
+                    let resolved_item = match resolve(
+                        NameKind::Term,
+                        &self.globals,
+                        self.locals.get_scopes(&self.curr_scope_chain),
+                        &item.name,
+                        &item.namespace,
+                    ) {
+                        Ok(res) => res,
+                        Err(_) => {
+                            // try to see if it is a type
+                            match resolve(
+                                NameKind::Ty,
+                                &self.globals,
+                                self.locals.get_scopes(&self.curr_scope_chain),
+                                &item.name,
+                                &item.namespace,
+                            ) {
+                                Ok(res) => res,
+                                Err(err) => {
+                                    self.errors.push(err);
+                                    continue;
+                                }
+                            }
+                        }
+                    };
+
```

Comment:
Shouldn't you just call `resolve_path` here (first with `NameKind::Ty`, then with `NameKind::Term`)? It does all the work for you (and some good error checking)